src/background/service-worker.ts

```
@@ -127,25 +94,29 @@ function _applyClientSideFilters(jobs) {
+  fetchedJobs: ProcessedJob[],
+  historicalSeenJobIds: Set<string>,
+  deletedJobIds: Set<string>,
+  currentCollapsedJobIds: Set<string>
+): Promise<{
+  allNewOrUpdatedJobsCount: number;
+  notifiableNewJobsCount: number;
+  updatedCollapsedJobIds: Set<string>;
+}> {
   // Filter out jobs that are already seen OR have been explicitly deleted by the user from the *fetched* list
   const allNewOrUpdatedJobs = fetchedJobs.filter(
     (job) => job && job.id && !historicalSeenJobIds.has(job.id) && !deletedJobIds.has(job.id)
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)